### PR TITLE
Add openshiftmasternodelabeler controller

### LINKED ISSUE #4168: Set up openshift 4 networking
**User Story**

As a user of openshift, I want funcitoning network

**Acceptance criteria**

* Pod to pod communication in an openshift cluster works
* Pod to service communcation in an openshift cluster works
* Node to pod communication in an openshift cluster works
* Node to service communication in an openshift cluster works

### PR DESCRIPTION
**Pull request by alvaroaleman**
**What this PR does / why we need it**:

This pr adds the `openshiftmasternodelabeler `, which is a simplistic controller that makes sure there are always three nodes labeled as master in an openshift cluster. This is necesary for network to function, because the network-operator creates a `sdn-controller` Daemonset which has a labelselector for master needs.

I manually verififed that with this changes here and a machine-controller version from master all of:
* Pod to Pod on same node
* Pod to Pod on different node
* Pod to service
* Node to Pod
* Pod to node

work.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #4168 

**Special notes for your reviewer**:

**Documentation**:

**Does this PR introduce a user-facing change?**:

```release-note
none
```
/approve

### CODE REVIEW

File: api/cmd/user-cluster-controller-manager/main.go [hunk 1]
```diff
@@ -199,6 +201,9 @@ func main() {
 		if err := nodecsrapprover.Add(mgr, 4, cfg); err != nil {
 			glog.Fatalf("failed to add nodecsrapprover controller: %v", err)
 		}
+		if err := openshiftmasternodelabeler.Add(context.Background(), kubermaticlog.Logger, mgr); err != nil {
+			glog.Fatalf("failed to add openshiftmasternodelabeler contorller: %v", err)
```
Comment (xrstf) [inline api/cmd/user-cluster-controller-manager/main.go]: Care to make this and the previous fatal errors begin with an uppercase latter?

File: api/pkg/controller/user-cluster-controller-manager/openshift-master-node-labeler/openshiftmasternodelabeler.go [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+	}
+
+	return nil
+}
+
+func (r *reconciler) Reconcile(_ reconcile.Request) (reconcile.Result, error) {
+	r.log.Info("Reconciling")
+	result, err := r.reconcile()
+	if err != nil {
+		r.log.Errorw("failed to reconcile", zap.Error(err))
```
Comment (xrstf) [inline api/pkg/controller/user-cluster-controller-manager/openshift-master-node-labeler/openshiftmasternodelabeler.go]: Uppercase error message please.

File: api/pkg/controller/user-cluster-controller-manager/openshift-master-node-labeler/openshiftmasternodelabeler.go [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+	nodesWithoutMasterLabel := sets.String{}
+	for _, node := range nodes.Items {
+		if _, exists := node.Labels["node-role.kubernetes.io/master"]; exists {
+			numNodesWithMasterLabel++
+		} else {
+			nodesWithoutMasterLabel.Insert(node.Name)
+		}
+	}
+
+	if numNodesWithMasterLabel >= 3 {
```
Comment (xrstf) [inline api/pkg/controller/user-cluster-controller-manager/openshift-master-node-labeler/openshiftmasternodelabeler.go]: Can you put this `3` and the `3` inside the following for-loop into a const like `minMasterNodes`?

**Comment by alvaroaleman** [comment]
@xrstf Updated, PTAL

**Pull request merged**